WordPress/gutenberg
# Rename core-data/src/selectors.js to selectors.ts (TypeScript)

## What?
This is a subset of https://github.com/WordPress/gutenberg/pull/39025 – a mega branch that proposes TypeScript signatures for all @wordpress/core-data selectors.

This PR renames `selectors.js` to `selectors.ts` and adjusts the related references in the documentation. It is the first step towards adding function signatures to these selectors. 

## Testing Instructions

Confirm the checks are green. The changes here should only affect the type system so there is nothing to test in the browser.

cc @dmsnell @jsnajdr @youknowriad @sarayourfriend @getdave @draganescu @scruffian
